editor/block-mover/style.scss

```diff
@@ -3,7 +3,6 @@
 	top: 10px;
 	left: 0;
 	padding: 0 14px 20px 0; // handles hover area
-	z-index: z-index( '.editor-block-mover' );
```

Comment:
There's a few related issues at #38 and #1511

My understanding is the general sentiment is that drag-and-drop is not a goal for primary block arrangement, but could be added as an enhancement on top of current arrangement tools. Drag-and-drop is tough to get right, and particularly so for long-content posts.

I don't have much personal experience with the two options you raised, aside from general awareness of their existing. I did have a hand in implementing [a similar component in Calypso](https://github.com/Automattic/wp-calypso/tree/master/client/components/forms/sortable-list) which could similar apply or inspire for a similar behavior in Gutenberg.